# Final Project Guidelines

The course includes a project worth 25% of the grade of the class. The project needs to be executed in a team consisting of 3-4 students. There are 5 deliverables as follows:

    Project proposal: 5% of the grade (.pdf file; team submission) (GradeScope)
    项目提案：成绩的 5%（.pdf 文件;团队提交）（GradeScope）
    Peer review of project proposal: 2% of the grade (individual submission) (Google Form)
    项目提案的同行评审：成绩的 2%（个人提交）（Google 表格）
    Final Presentation: 5% of the grade (.pdf file; team submission) (GradeScope)
    最终 演示：成绩的 5%（.pdf 个文件;团队提交）（GradeScope）
    Final report & code: 10% of the grade (code & .pdf file; team submission) (GradeScope)
    最终报告和代码：成绩的10%（代码和.pdf文件;团队提交）（GradeScope）
    Peer review of final report and presentation: 3% of the grade (individual submission) (Google Form)
    最终报告和演示文稿的同行评审：成绩的 3%（个人提交）（Google 表格）

Project description

    Analyze a biological dataset of your choice. Complex datasets (high-dimensional, large-scale, unstructured noise, etc.) increase novelty.
    Apply 3 ML methods for 3 ML tasks (e.g., K-mean for clustering, PCA for dimensionality reduction). Adequate justification of methods and tasks chosen merit higher grades.
    将 3 个 ML 方法应用于 3 个 ML 任务（例如，用于聚类的 K-mean，用于降维的 PCA）。所选方法和任务的充分理由值得更高的分数。
    You need to implement at least 2 out of 3 methods yourselves and may use existing packages for at most 1 out 3 methods.
    您需要自己实现 3 种方法中的至少 2 种，并且最多可以为 3 种方法中的 1 种使用现有包。
    Provide details for parameter tuning (e.g., using cross-validation). More comprehensive tuning of methods merit higher grades.
    提供参数优化的详细信息（例如，使用交叉验证）。方法的调整越全面，评分越高。
    You need to clearly define evaluation metrics and present results (e.g., F1 score for classification accuracy). More comprehensive evaluations merit higher grades.
    您需要明确定义评估指标并呈现结果（例如，分类准确性的 F1 分数）。更全面的评估值得更高的分数。
    You need to submit the code along with the project report. Better documentation & reproducibility merit higher grades.
    您需要将代码与项目报告一起提交。更好的文档和可重复性值得更高的等级。

Timeline

1.31 - Finalize project teams

2/3-2/17 - 15min meetings to discuss project ideas with Prof. Xu or Prof. Zhang (voluntary)
2/3-2/17 - 15 分钟会议，与 Xu 教授或 Zhang 教授讨论项目想法（自愿）

3/14 - Project proposal due
3/14 - 项目提案到期

3/21 - Peer review on project proposal due
3/21 - 项目提案的同行评审到期

4/21-25 - Project presentation
4/21-25 - 项目介绍

4/30 - Final report due
4/30 - 最终报告到期

5/2 - Peer review on final report and presentation due
5/2 - 对最终报告和演示文稿进行同行评审
Deliverables 

Detailed instructions on the deliverables are provided below:

    Oral presentation 

    Time: 4/21-4/25, in-class
    时间：4/21-4/25，课堂
    Around 5-10-minutes for each group, including Q&A (TBD).
    The presentation slides will be submitted Monday, April 21, before class at 8:00 AM (.pdf format).
    演示幻灯片将于 4 月 21 日星期一上午 8：00 课前提交（.pdf格式）。
    Presentation rubric

    Background (2%): motivation, background, and related works should be clearly explained.
    Analysis (2%): analysis, simulation procedures, and data should be clearly described.
    Results (2%): the results should be clearly presented using appropriate graphics.
    结果 （2%）：结果应使用适当的图形清晰呈现。
    Conclusions (2%): the conclusions should be reasonable based on the data and evidence presented.
    结论 （2%）：根据所提供的数据和证据，结论应该是合理的。
    Questions (2%): students answer all class questions with explanations and elaboration.
    问题 （2%）：学生通过解释和阐述回答所有课堂问题。

    Final report & code Each team will submit a final report. A six-page report with additional pages for references in the Neurips format (https://neurips.cc/Conferences/2023/PaperInformation/StyleFiles). The final report should have the following structure. The code should be accompanied with instructions on how to execute them and reproduce main results of the paper.
    最终报告和代码 每个团队都将提交一份最终报告。 一份六页 的报告，其中包含额外的页面，用于 Neurips 格式的参考 （https://neurips.cc/Conferences/2023/PaperInformation/StyleFiles）。最终报告应具有以下结构。该代码应附有有关如何执行它们和重现论文主要结果的说明。

(a) Introduction (one paragraph on each item below)
（a） 引言（以下各项各一段）

•   State the problem you are trying to address.

•   Provide the motivation.
•提供动力。

•   Provide a high-level overview of the three methods that you used to analyze the data.
•提供用于分析数据的三种方法的高级概述。

•   Discuss the dataset you analyzed
•讨论您分析的数据集

(b) Data

•  Provide details about data. You should specify number of samples, number of features, details about features(are they all continuous variables?, which information can we obtain from features?
• 提供有关数据的详细信息。您应该指定样本数、特征数、特征的详细信息（它们都是连续变量吗？，我们可以从特征中获得哪些信息？

(c) Methods

•   As a team, we ask you to apply at least three different methods on the dataset of your choice. You should implement two of the three on your own. For the other one, you are welcome to use an existing package. You are welcome to try out other packages in addition to the required minimum three methods. Describe the three methods you applied to the data in detail, one in each subsection.
•   作为一个团队，我们要求您在您选择的数据集上应用至少三种不同的方法。您应该自己实现这三个选项中的两个。对于另一个，欢迎使用现有包。除了所需的最少三种方法外，欢迎您尝试其他软件包。详细描述您对数据应用的三种方法，每个小节中各一种。

(d) Results

•   Provide results as figures and tables. Each figure and table should have a caption.
•以图形和表格的形式提供结果。每个图形和表格都应该有一个标题。

•   In the text, explain your findings in figures. In each paragraph, discuss how each result was generated, make observations from the results in the figure, and draw mini conclusions.
•在文本中，用数字解释您的发现。在每个段落中，讨论每个结果是如何产生的，从图中的结果中进行观察，并得出小结论。

(e) Discussions

•   Summarize your findings.
•总结您的发现。

•   Discuss future work.
•讨论未来的工作。

Use the above as a checklist. Create a section for each of (a)-(e) above. This is a typical structure of a scientific paper. Your grade will reflect how closely you follow this structure.
使用上述内容作为检查清单。为上述 （a）-（e） 分别创建一个部分。这是科学论文的典型结构。您的成绩将反映您遵循此结构的程度。

What to submit For the final report submission, please submit the report itself (.pdf file), the codes of the three methods you wrote with brief instructions on how to run them, and the slides from the presentation (.pdf format).
提交内容 对于最终报告提交，请提交报告本身（.pdf文件）、您编写的三种方法的代码以及有关如何运行它们的简要说明，以及演示文稿中的幻灯片（.pdf格式）。

    Project proposal: You can think of the project proposal as a short draft of the introduction of the final report. You will further revise this before submitting the final report at the end of the semester. The proposal should include the title, team members as authors, and the introduction section. We recommend the proposal be limited to one page, but this is not a strict requirement. The proposal should contain the followings:
    项目提案：您可以将项目提案视为最终报告介绍的简短草稿。在学期末提交最终报告之前，您将进一步修改此报告。提案应包括标题、作为作者的团队成员和引言部分。我们建议将提案限制为一页，但这并不是一个严格的要求。提案应包含以下内容：
    The problem or question you want to address
    您要解决的问题
    A short summary of the related works (at least 2)
    相关作品的简短摘要（至少 2 篇）
    An overview of the three methods/ideas you plan to implement
    您计划实施的三种 方法/想法 的概述
    An explanation of how you will evaluate the performance of your methods
    关于您将如何评估方法性能的说明
    Review of Project Proposal, Presentation, and Final Report: Each student is required to review 3 proposals, 3 final presentations, and 3 final reports via separate google forms. We will assign the review assignments after the proposal submission.
    审查项目提案、演示文稿和最终报告： 每个学生都需要通过单独的 google 表格审查 3 份提案、3 份最终报告和 3 份最终报告。我们将在提案提交后分配审核任务。

    Example Google form for submitting reviews

    https://docs.google.com/forms/d/e/1FAIpQLSen2PKigAuElGaNMtRrtBsu_cNknYYFk96FdYPT9vIemLFqYw/viewform?usp=dialog

Suggested Datasets

You can also use your own real-world datasets (not simulated data). Using complex datasets will increase the novelty of your project.  

    [Bulk RNA-Seq] Genotype-Tissue Expression (GTEx) data (https://www.gtexportal.org/home/): Human gene expression data for many different tissue types. You can download the gene expression data from the web portal. Note that you cannot access the genetic data, as this requires an approval. You could perform unsupervised learning (clustering or network estimation) on the data from a single tissue type. You could compare your findings across multiple tissue types.
    [Bulk RNA-Seq] 基因型-组织表达 （GTEx） 数据 （https://www.gtexportal.org/home/）：许多不同组织类型的人类基因表达数据。您可以从 Web 门户下载基因表达数据。请注意，您无法访问遗传数据，因为这需要批准。您可以对来自单个组织类型的数据执行无监督学习（聚类或网络估计）。您可以比较多种组织类型的发现。

    The GTEx Consortium. The Genotype-Tissue Expression (GTEx) pilot analysis: Multitissue gene regulation in humans. Science, 2015.
    GTEx 联盟。基因型-组织表达 （GTEx） 试点分析：人类多组织基因调控。科学，2015 年。

    [Single-cell RNA-seq] Tabula Muris single-cell gene expression data. Mouse single-cell RNA-seq data across tissues and organs. This is similar to GTEx data above, but was collected at the single-cell level. This dataset contains cell type labels as well as age of mice. So in addition to the unsupervised learning tasks for the GTEx data above, you can also perform supervised learning (e.g., predicting cell type / age of cells). Single-cell data has a large number of missing data points, so you will have to impute the missing values for analysis.
    [单细胞 RNA-seq]Tabula Muris 单细胞基因表达数据。跨组织和器官的小鼠单细胞 RNA-seq 数据。这与上面的 GTEx 数据类似，但是在单细胞级别收集的。此数据集包含细胞类型标签以及小鼠的年龄。因此，除了上述 GTEx 数据的无监督学习任务外，您还可以执行监督学习（例如，预测细胞类型/细胞年龄）。单个单元格数据具有大量缺失数据点，因此您必须插补缺失值才能进行分析。

    Data portal https://tabula-muris.ds.czbiohub.org/
    "A single-cell transcriptomic atlas characterizes ageing tissues in the mouse." Nature 583, no. 7817 (2020): 590-595.
    “单细胞转录组图谱表征了小鼠的衰老组织。”自然 583，第 7817 期（2020 年）：590-595。
    Tutorial for processing single-cell data (you can also use the PBMC3K data in this tutorial): https://scanpy-tutorials.readthedocs.io/en/latest/pbmc3k.html 

    [Single-cell RNA-seq] Human Cell Atlas (https://www.humancellatlas.org/): Human single-cell RNA-seq data for various organs are available in the data portal. This is similar to GTEx data above, but was collected at the single-cell level. A similar type of analysis described above for GTEx data could be done. Unlike the GTEx data, this has a large number of missing data points, so you will have to impute the missing values for analysis.
    [单细胞 RNA-seq]人类细胞图谱 （https://www.humancellatlas.org/）：数据门户中提供了各种器官的人类单细胞 RNA-seq 数据。这与上面的 GTEx 数据类似，但是在单细胞级别收集的。可以对 GTEx 数据进行上述类似类型的分析。与 GTEx 数据不同，它有大量缺失数据点，因此您必须插补缺失值才能进行分析。

    Rozenblatt-Rosen et al. Building a high-quality Human Cell Atlas. Nature Biotechnology, 2021.

    The Cancer Genome Atlas (TCGA) Data (https://portal.gdc.cancer.gov/repository): This is a consortium data with various types of genomic data collected from a large number of patients for many different cancer types. Gene expression, copy number variation, and epigenetic, clinical data are available for open access. You could perform unsupervised learning on a single type of data for different cancer types. You could perform supervised learning on two types of data. For example, you could predict gene expression levels based on copy number variation, or predict clinical outcome based on gene expression and/or copy number variation.
    癌症基因组图谱 （TCGA） 数据 （https://portal.gdc.cancer.gov/repository）：这是一个联盟数据，其中包含从大量患者那里收集的各种类型的基因组数据，适用于许多不同癌症类型。基因表达、拷贝数变异和表观遗传学、临床数据可供开放获取。您可以对不同癌症类型的单一数据类型执行无监督学习。您可以对两种类型的数据执行监督式学习。例如，您可以根据拷贝数变异预测基因表达水平，或根据基因表达和/或拷贝数变异预测临床结果。

    Hutter et al. The Cancer Genome Atlas: creating lasting value beyond its data. Cell, 2018.

    [Genotype and gene expression] Mouse advanced intercross data (http://palmerlab.org/protocols-data/): The expression data appeared in Homework 2. Data are available for both brain gene expression and genotypes for the mouse population after 50 generations of inbreeding between two inbred founder mice. You could perform supervised learning to predict gene expression levels based on the genetic data.
    [基因型和基因表达] 小鼠高级杂交数据 （http://palmerlab.org/protocols-data/）：表达数据出现在 Homework 2 中。在两只近交创始小鼠之间近亲繁殖 50 代后，小鼠种群的大脑基因表达和基因型数据可用。您可以执行监督学习，以根据遗传数据预测基因表达水平。

    Gonzeles et al. Genome-wide association analysis in a mouse advanced intercross line. Nature Communications, 2018.
    Gonzeles 等人。小鼠高级交叉系中的全基因组关联分析。Nature Communications，2018 年。

    Allen Brain Map (https://portal.brain-map.org/): Various data types from human and mouse brains are available.
    Allen Brain Map （https://portal.brain-map.org/）：来自人类和小鼠大脑的各种数据类型可用。

    Shen et al. The Allen Human Brain Atlas: Comprehensive gene expression mapping of the human brain. Trends in Neuroscience, 2012.
    Shen 等人。Allen Human Brain Atlas：人脑的综合基因表达图谱。神经科学趋势，2012 年。
    Jones et al. The Allen Brain Atlas: 5 years and beyond, Nature Reviews, 2009.
    Jones 等人。艾伦大脑图谱：5 年及以后，自然评论，2009 年。

Other suggestions

    In all data analysis projects, one should take time to learn about the data. Go to the data portal and spend some time browsing the data. Go to scholar.google.com and search using the name of the dataset above as a keyword. You will find many papers published for the given dataset that you may find useful.
    在所有数据分析项目中，都应该花时间了解数据。转到数据门户并花一些时间浏览数据。转到 scholar.google.com 并使用上述数据集的名称作为关键字进行搜索。您将找到许多针对给定数据集发表的论文，您可能会觉得这些论文很有用。
    In case you face any trouble with Latex, you may follow the Latex tutorial The (not so) short introduction to Latex. https://gking.harvard.edu/files/ lshort2.pdf
    如果您在使用 Latex 时遇到任何问题，您可以按照 Latex 教程 Latex 的（不是那么）简短介绍。https://gking.harvard.edu/files/ lshort2.pdf
    Know your data. Download your data and explore it as early as possible. What's the number of samples and the number of features? How many NAN values? Does it have labels? What are the preprocessing steps?
    了解您的数据。下载您的数据并尽早进行探索。样本数和特征数是多少？有多少个 NAN 值？它有标签吗？预处理步骤有哪些？
    Don't be too ambitious. You will encounter many problems when you work on real data. High ambition --> more frustration. Clearly define 3 ML goals and provide careful evaluation. This is a course project, not curing cancer.
    不要太雄心勃勃。在处理真实数据时，您会遇到很多问题。雄心勃勃 --> 更受挫。明确定义 3 个 ML 目标并提供仔细评估。这是一个课程项目，而不是治愈癌症。
    What if your method doesn't work? Have a backup plan. If your method 2 and method 3 depend on the results of method 1, that's really risky. What if method 1 doesn't work? Also, try to test your ideas using existing software and make sure they work, before spending time on your own implementation.
    如果您的方法不起作用怎么办？制定备份计划。如果你的方法 2 和方法 3 依赖于方法 1 的结果，那真的很有风险。如果方法 1 不起作用怎么办？此外，在花时间进行自己的实施之前，请尝试使用现有软件测试您的想法并确保它们有效。
    Devil is in the details. 3 carefully executed analyses >> 10 superficial analyses. Did you choose the appropriate methods and why? How did you determine the parameters? Did you do cross-validation? Is your evaluation criteria fair?
    魔鬼在细节中。3 个仔细执行的分析 >> 10 个肤浅的分析。您是否选择了合适的方法，为什么？您是如何确定参数的？您是否进行了交叉验证？您的评估标准公平吗？

